main.py

```diff
@@ -246,26 +252,44 @@ def match_flows(self, switch, args, stored_flows, many=True):
             return None
         return response
 
-    # pylint: disable=redefined-outer-name
-    # pylint: disable=too-many-branches
+    def get_action(self, switch, table_id, args, stored_flows, actions):
+        """Get actions in the matched flow"""
+        goto_table = False
+        actions_ = []
+        flow = self.match_flows(switch, table_id, args, stored_flows, False)
+        if flow and 'actions' in flow['flow']:
+            actions_ = flow['flow']['actions']
+        elif flow and 'instructions' in flow['flow']:
+            for instruction in flow['flow']['instructions']:
+                if instruction['instruction_type'] == 'apply_actions':
+                    actions_ = instruction['actions']
+                elif instruction['instruction_type'] == 'goto_table':
+                    table_id_ = instruction['table_id']
+                    if table_id < table_id_:
+                        table_id = table_id_
+                        goto_table = True
+                    else:
+                        log.error(f"A packet can only been directed to a \
+                                  flow table number greather than {table_id}")
+        actions.extend(actions_)
+        return flow, actions, goto_table, table_id
+
     def match_and_apply(self, switch, args, stored_flows):
-        # pylint: disable=bad-staticmethod-argument
         """Match flows and apply actions.
         Match given packet (in args) against
         the stored flows (from flow_manager) and,
         if a match flow is found, apply its actions."""
-        flow = self.match_flows(switch, args, stored_flows, False)
+        table_id = 0
+        goto_table = True
         port = None
         actions = []
-        # pylint: disable=too-many-nested-blocks
+        while goto_table:
```

Comment:
You're implementation of `goto_table` is not resulting in a new trace_step result, which means whoever is reading each step has no idea which table it's on, which might be OK, it's recursively resolving the table in that step, although it'd be less flexible to also use it for debugging and tracing the tables. Currently, `sdntrace_cp` being used by `mef_eline` so this initial implementation can work too, but again, it'd be less flexible for other usages of `sdntrace_cp`. Please sync with @italovalcy and @jab1982 if they'd like to see each table on each result step or not.

Reply:
> You're implementation of `goto_table` is not resulting in a new trace_step result, which means whoever is reading each step has no idea which table it's on, which might be OK, it's recursively resolving the table in that step, although it'd be less flexible to also use it for debugging and tracing the tables. Currently, `sdntrace_cp` being used by `mef_eline` so this initial implementation can work too, but again, it'd be less flexible for other usages of `sdntrace_cp`. Please sync with @italovalcy and @jab1982 if they'd like to see each table on each result step or not.

@viniarck that is a very interesting point and for sure we will get into this point soon. I would say that even though it would be very interesting to see detailed information, maybe we can leave as it is for now and we open a issue for enhance this later on, when a more specific use case shows up. Wdyt?

Reply:
Sounds good to me too, also at least all out going ports are expected to show up so for `telemetry_int` the external loops will show up too. @gretelliz, if you can map an issue for this without the `2023.2` for now.